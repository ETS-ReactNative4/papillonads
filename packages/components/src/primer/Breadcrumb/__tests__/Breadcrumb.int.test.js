import { ariaAttr, items, renderCustom, renderDefault } from './Breadcrumb.int.render'

jest.mock('@papillonads/library', () => {
  return {
    hooks: {
      react: {
        useState: () => {},
      },
    },
    array: {
      getIndexItems: () => {},
      getIndexItemsWithSelected: () => {},
    },
  }
})

const libraryMockObject = require('@papillonads/library')

describe('<Breadcrumb />', () => {
  let indexItemsDataObject = [
    { ...items[0], index: 0 },
    { ...items[1], index: 1 },
    { ...items[2], index: 2 },
  ]

  const newIndexItemsWithSelectedDataObject = [
    { ...items[0], index: 0, isSelected: true },
    { ...items[1], index: 1, isSelected: false },
    { ...items[2], index: 2, isSelected: false },
  ]

  const newItemsDataObject = [
    { ...items[0], isSelected: true },
    { ...items[1], isSelected: false },
    { ...items[2], isSelected: false },
  ]

  const onClickMockFn = jest.fn()

  const setIndexItemsMockFn = jest.fn((indexItems) => {
    indexItemsDataObject = indexItems
  })

  beforeEach(() => {
    jest.spyOn(libraryMockObject.hooks.react, 'useState').mockImplementation(() => {
      return [indexItemsDataObject, setIndexItemsMockFn]
    })

    jest.spyOn(libraryMockObject.array, 'getIndexItems').mockReturnValue(indexItemsDataObject)

    jest.spyOn(libraryMockObject.array, 'getIndexItemsWithSelected').mockReturnValue(newIndexItemsWithSelectedDataObject)
  })

  afterEach(() => jest.clearAllMocks())

  describe('Event', () => {
    test('must return new items without index when onClick()', () => {
      const mountCustomRender = global.renderMount(renderCustom(onClickMockFn))
      mountCustomRender.find('li').first().simulate('click')
      expect(mountCustomRender.props().onClick).toBe(onClickMockFn)
      expect(mountCustomRender.props().onClick).toHaveBeenCalledWith({
        active: newItemsDataObject[0].text,
        ariaAttr,
        items: newItemsDataObject,
      })
    })
  })

  describe('Render', () => {
    test('must match renderDefault()', () => {
      expect(global.renderToJSON(renderDefault())).toMatchSnapshot()
    })
  })
})
