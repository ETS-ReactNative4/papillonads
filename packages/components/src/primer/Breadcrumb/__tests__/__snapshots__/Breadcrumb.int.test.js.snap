// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> Render must match renderDefault() 1`] = `
<nav
  aria-label="Breadcrumb"
  className=""
>
  <ol>
    <li
      className="breadcrumb-item"
      href="/fantastic-folder"
      onClick={[Function]}
    >
      <a
        href="/fantastic-folder"
      >
        Fantastic Folder
      </a>
    </li>
    <li
      className="breadcrumb-item"
      href="/fantastic-folder/another-fantastic-folder"
      onClick={[Function]}
    >
      <a
        href="/fantastic-folder/another-fantastic-folder"
      >
        Another Fantastic folder
      </a>
    </li>
    <li
      aria-current="page"
      className="breadcrumb-item breadcrumb-item-selected text-gray"
      href=""
      onClick={[Function]}
    >
      One More Fantastic Folder
    </li>
  </ol>
</nav>
`;
